src/test/opentrackIO_parser/python/opentrackIOlib.py

```diff
@@ -0,0 +1,304 @@
+            denominator = float(self.pd["timing"]["timecode"]["format"]["frameRate"]["denom"]) 
+            return float(numerator / denominator)
+        else:
+            return None
+
+    def Set_translation_units(self,unit_str):
+        """Establish user-preferred units for translations. 
+        Valid args: m, cm, mm, in"""
+        schema_units = self.sd["properties"]["transforms"]["items"]["items"]["properties"]["translation"]["units"]
+        if self.verbose:
+            print("Schema says camera translation units are: {}".format(schema_units))
+            print("Setting preferred translation units to: {0}".format(unit_str))
+        if schema_units == "meter":
+            self.trans_mult = self.Conversion_factor_from_meters(unit_str)
+    
+    def Set_rotation_units(self,unit_str):
+        """Establish user-preferred units for rotations. 
+        Valid args: deg, rad"""
+        schema_units = self.sd["properties"]["transforms"]["items"]["items"]["properties"]["rotation"]["units"]
+        if self.verbose:
+            print("Schema says camera rotation units are: {}".format(schema_units))
+            print("Setting preferred camera rotation units to: {0}".format(unit_str))
+        if schema_units == "degree":
+            if unit_str == "deg":
+                self.rot_mult = 1.0
+            if unit_str == "rad":
+                self.rot_mult = math.pi/180
+
+    def Set_sample_time_format(self,format_str):
+        """Establish user preference for sample time format.
+        Valid args: sec, timecode, string"""
+        if self.verbose:
+            schema_units = self.sd["properties"]["timing"]["properties"]["sampleTimestamp"]["units"]
+            print("Schema says sample time units are: {}".format(schema_units))
+            print("Setting preferred sample time format to: {0}".format(format_str))
+        if (format_str in self.sample_time_formats):
+            self.sample_time_format = format_str
+
+    def Set_focus_distance_units(self,unit_str):
+        """Establish a user-preference for units of focus distance by storing a conversion factor.
+        Valid: m, cm, mm, in"""
+        schema_units = self.sd["properties"]["lens"]["properties"]["focusDistance"]["units"]
+        if self.verbose:
+            print("Schema says focus distance units are: {}".format(schema_units))
+            print("Setting preferred focus distance units to: {}".format(unit_str))
+        if schema_units == "millimeter":
+            if unit_str == "m":
+                self.focus_dist_mult = .001
+            elif unit_str == "cm":
+                self.focus_dist_mult = 0.1
+            elif unit_str == "mm":
+                self.focus_dist_mult = 1.0
+            elif unit_str == "in":
+                self.focus_dist_mult = 1/25.4
+
+    def Get_protocol(self):
+        """The protocol to which this sample conforms"""
+        #if (self.pd["protocol"]):
```

Comment:
Can now be re-added if you merge my latest commit

Reply:
Done